kdbg/debugger.cpp

```diff
@@ -2155,30 +2159,43 @@ void KDebugger::slotDisassemble(const QString& fileName, int lineNo)
 
 void KDebugger::submitDisassemblyFlavor()
 {
+    /*
+     * If m_flavor is empty (ie this program is loaded for the first time),
+     * read m_globalFlavor and act accordingly.
+     */
+    if (m_flavor.isEmpty()) {
+	if (m_globalFlavor.isEmpty()
+	    || m_globalFlavor == DefaultGenericFlavor
+	    || m_globalFlavor == DefaultX86Flavor) {
+	    m_flavor = DefaultX86Flavor;
+	} else {
+	    m_flavor = m_globalFlavor;
+	}
+    }
+
     m_d->executeCmd(DCsetdisassflavor, m_flavor);
 }
 
-void KDebugger::setDefaultFlavor(QString defFlavor)
+void KDebugger::setDefaultFlavor(QString defFlavor, bool updateView)
 {
-    m_globalFlavor = defFlavor;
-
     /*
-     * To specify a flavor for this program first read the global default.
-     * debugProgram will later set the (global) flavor. The user can later
-     * override the global flavor for this program. If the user reloaded a
-     * binary, `reloaded` is true and the current flavor is set forcefully
-     * to the global.
+     * Global default uses "default" while This Program uses
+     * "use global setting". This leads to the wrong string being
+     * stored for This program's configuration and broken logic
+     * for m_flavor. Make sure that when "default" is globally
+     * selected, KDebugger will see "use global setting" for it's
+     * purposes.
      */
-    if (m_globalFlavor.isEmpty() && m_flavor.isEmpty()) {
-	m_flavor = DefaultDbgX86Flavor;
-    } else if (m_flavor.isEmpty() ) {
-	m_flavor = m_globalFlavor;
+    if (defFlavor == DefaultGenericFlavor) {
+	defFlavor = DefaultDbgGenericFlavor;
     }
+    m_globalFlavor = defFlavor;
 
     /*
-     * Flavor can't be submitted to NULL debugger pointer
+     * Check if we should update the view. Also flavor can't be submitted
+     * to NULL debugger pointer.
      */
-    if(m_d != 0){
+    if (updateView && m_d != 0) {
```

Comment:
As said elsewhere, I cannot follow the logic why sometimes an update should not happen.

Regarding the comment:

- Please do not mention trivial things like "can't be submitted to NULL debugger pointer"; such a check is just good programming habit; it's not noteworthy.
- Do not say "Check if we should..." if there is no check. The fact that we inspect a variable is not "a check". If that were a more complicated expression, then a comment could be helpful. As the `if`-statement is currently written, it is as natural language as can be, and does not require a comment.

Reply:
> As said elsewhere, I cannot follow the logic why sometimes an update should not happen.

Because changing global flavor shouldn't take effect at the moment. At least I thought so. I will make global flavor able to perform an update if This program's flavor is set to default.